### ISSUE TITLE
panos_ike_gateway -> protocol-common -> nat-traversal is invalid

### ISSUE THREAD

**Issue by monemhamid** [terraform-providers/terraform-provider-panos#97]
while setting up provisioning for pan through terraform, the nat traversal boolean "true" is not accepted by the firewall

the error is 

panos_ike_gateway.gw_name:  gw_name -> protocol-common -> nat-traversal is invalid

**Comment by btorresgil** [comment]
Thanks for opening this issue.  We'll look into this.

**Comment by shinmog** [comment]
@monemhamid 

There are a few options that are under that xpath, but the resource is working for me, and I can create an IKE gateway with `enable_nat_traversal` set to `True`:

```hcl
resource "panos_ike_gateway" "gw" {
    name = "chk97"
    version = "ikev1"
    local_ip_address_type = "ip"
    local_ip_address_value = "10.2.1.1/24"
    interface = "ethernet1/8"
    peer_ip_type = "ip"
    peer_ip_value = "5.5.5.55"
    pre_shared_key = "secret"
    enable_nat_traversal = true
    nat_traversal_keep_alive = 20
    enable_dead_peer_detection = true
    dead_peer_detection_interval = 5
    dead_peer_detection_retry = 5
}
```

Can you please verify that your config works if you try to enter it into the GUI?  If it seems to be matching up and this is still failing, please share which version of PAN-OS you're running against as well as the Terraform plan file config for your IKE gateway.

**Comment by monemhamid** [comment]
Hello
Apologies for not responding in time
this is still not working for me. I double-checked, the same config in GUI is working
PAN-OS version is 8.1.0
TF file is below
```
resource "panos_ike_gateway" "a-to-b" {
  name                              = "a-to-b"
  local_id_value                    = "1.1.1.1"
  interface                         = "ethernet1/1"
  peer_ip_type                      = "ip"
  peer_ip_value                     = "2.2.2.2"
  pre_shared_key                    = "paloalto"
  local_id_type                     = "ipaddr"
  peer_id_type                      = "ipaddr"
  peer_id_value                     = "2.2.2.2"
  ikev2_crypto_profile              = "default"
  enable_nat_traversal              = "true"
  version                           = "ikev2"
  #enable_fragmentation             = "0"
  #liveness_check_interval          = "5"
  #enable_dead_peer_detection       = "true"
  #dead_peer_detection_interval     = "5"
  #dead_peer_detection_retry        = "5"
  enable_liveness_check             = "true"
  nat_traversal_keep_alive          = "5"
  #enable_passive_mode               = "false"
  #nat_traversal_enable_udp_checksum = "false"
}
```
this code runs fine as soon as i disable "enable_nat_traversal" and "nat_traversal_keep_alive" parameters. When i enable "enable_nat_traversal", i have to enable "nat_traversal_keep_alive" as well, otherwise it gives an error 
```
* panos_ike_gateway.a-to-b: 1 error(s) occurred:

* panos_ike_gateway.a-tob:  a-to-b -> protocol-common -> nat-traversal -> keep-alive-interval is invalid
```
and finally, when both parameters are enable, i get the error
```
* panos_ike_gateway.a-to-b: 1 error(s) occurred:

* panos_ike_gateway.a-to-b:  a-to-b -> protocol-common -> nat-traversal is invalid
```

**Comment by monemhamid** [comment]
i updated the fw code to 8.1.5, the issue is still there

**Comment by fluffy-cakes** [comment]
any update on this? i have the very same issue.

**Comment by shinmog** [comment]
@monemhamid @fluffy-cakes 

After looking into this, it seems that the field `nat_traversal_keep_alive` has a few stipulations:

1) 10 <= x <= 3600
2) must be a multiple of 10

So the reason it worked for me is because I tried `20` in my example which meets to two criteria.

Sorry it took so long to figure this out.  I ended up having to brute force that second rule out of PAN-OS.  So, try giving it a value that adheres to the two above limitations and let me know if it works for you.  If so, I'll need to update the documentation to mention this.
